resources/agent.rb

```diff
@@ -0,0 +1,66 @@
+#
+# Cookbook Name:: sc-mongodb
+# Resource:: agent
+#
+# Copyright 2017, Grant Ridder
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+provides :mongodb_agent
+
+property :type, String, name_property: true, equal_to: %w(automation backup monitoring)
+
+property :config, Hash
+property :group, String
+property :package_url, String
```

Comment:
Ah, this alleviates my version worries. Also makes life better for folks who host their own packages.